### PR TITLE
refactor: make use of Babel

### PR DESCRIPTION
**Pull request by hiroppy**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [x] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

We use babel to the core code so I added a `build` task and rewrote CJS to ESM.

### Breaking Changes

no

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@ Thumbs.db
 .vscode
 *.sublime-project
 *.sublime-workspace
+lib
```
[the remark this answers is not in the document]
Comment (hiroppy) [reply .gitignore]: No, currently we transpile from `src` to `lib` so we need to add `lib`.

File: package.json [hunk 2]
```diff
@@ -63,6 +65,7 @@
     "lint-staged": "^9.4.2",
     "npm-run-all": "^4.1.5",
     "prettier": "^1.18.2",
+    "rimraf": "^3.0.0",
```
[the remark this answers is not in the document]
Comment (hiroppy) [reply package.json]: deleted

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (evilebottnawi) [reply .gitignore]: Let's transient to dist, it is good practice

[the remark this answers is not in the document]
Comment (hiroppy) [reply .gitignore]: I think `lib` is better. For example, babel's code is transpiled from `src` to `lib`. https://github.com/babel/babel/blob/master/Gulpfile.js#L65-L75

[the remark this answers is not in the document]
Comment (hiroppy) [reply .gitignore]: Ummm, ok I'll change but I would like to say `lib` is also a good practice.

**Comment by hiroppy** [comment]
Updated

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (evilebottnawi) [reply .gitignore]: I really think it’s not so important, we can always change it

**Comment by hiroppy** [comment]
@evilebottnawi Could you approve this?

**Comment by hiroppy** [comment]
Thank you for reviewing, I'll merge~

**Pull request merged**